### PR TITLE
CEDS-1055: moved inventory linking api call to this service and refactored code so 

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
controller takes payload

### CODE REVIEW

File: app/uk/gov/hmrc/exports/movements/services/MovementsService.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    movementsRepo
+      .save(movementSubmission)
+      .map(
+        res =>
+          if (res) {
+            Logger.debug("movement submission data saved to DB")
+            true
+          } else {
+            Logger.error("error  saving movement submission data to DB")
+            false
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/services/MovementsService.scala]:
```
if(res) Logger.debug("movement submission data saved to DB")
else Logger.error("error  saving movement submission data to DB")
res
```
Just proposal, I don't like if returning boolean :D

File: precheck.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-sbt clean coverage test it:test scalafmt::test test:scalafmt::test coverageReport
+sbt clean coverage test it:test coverageReport
```
Comment (PatrykRudnicki) [inline precheck.sh]: Why you remove it? It's only checking and showing errors, we should have some style defined

**Comment by PatrykRudnicki** [comment]
And please run formatter, few methods have wrong indentation

File: precheck.sh (hunk 2 above)
Comment (mattclark-zerogravit) [reply precheck.sh]: Every time I run precheck to see if my test are passing i get 20+ changes that I have to revert... yes to scalafmt not in precheck...

File: app/uk/gov/hmrc/exports/movements/services/MovementsService.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/exports/movements/services/MovementsService.scala]: done

**Pull request merged**